Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -5068,4 +5068,40 @@ void Document::set_needs_to_refresh_scroll_state(bool b)
         paintable->set_needs_to_refresh_scroll_state(b);
 }
 
+Vector<JS::NonnullGCPtr<Range>> Document::find_matching_text(String const& query)
+{
+    if (!document_element() || !document_element()->layout_node())
+        return {};
+
+    Vector<JS::NonnullGCPtr<Range>> matches;
+    Function<void(Layout::Node const&)> recurse = [&](auto& node) {
+        for (auto const* child = node.first_child(); child; child = child->next_sibling()) {
```

Comment:
Could use `first_child_of_type<Layout::TextNode>` and `next_sibling_of_type<Layout::TextNode>` here and remove one level of indentation.

Reply:
I used `for_each_in_inclusive_subtree_of_type<Layout::TextNode>()` here.